File: 2025-09-16 - Lista Dup Encadeada/Exemplo_Java/ListaDuplaEncadeadaInterface.java
// ------------------------------------------------------------------------
// Interface ListaDuplaEncadeadaInterface
// Define as operações públicas esperadas para a lista duplamente encadeada.
// ------------------------------------------------------------------------

public interface ListaDuplaEncadeadaInterface {

   boolean inserirNoInicio(String codigo, String nome, int cargaHoraria);

   boolean inserirNoFim(String codigo, String nome, int cargaHoraria);

   boolean inserirNaPosicao(String codigo, String nome, int cargaHoraria, int posicao);

   boolean removerPorCodigo(String codigo);

   Node buscarPorCodigo(String codigo);

   boolean atualizarDisciplina(String codigo, String novoNome, int novaCarga);

   void imprimirLista();

   boolean estaVazia();

   int tamanho();
}

File: 2025-09-16 - Lista Dup Encadeada/Exemplo_C#/ListaDuplaEncadeadaInterface.cs
// ------------------------------------------------------------------------
// Interface ListaDuplaEncadeadaInterface
// Define os métodos públicos esperados
// ------------------------------------------------------------------------

using System;

namespace Estruturas
{
   public interface ListaDuplaEncadeadaInterface
   {
      bool InserirNoInicio(string codigo, string nome, int cargaHoraria);
      bool InserirNoFim(string codigo, string nome, int cargaHoraria);
      bool InserirNaPosicao(string codigo, string nome, int cargaHoraria, int posicao);
      bool RemoverPorCodigo(string codigo);
      Node BuscarPorCodigo(string codigo);
      bool AtualizarDisciplina(string codigo, string novoNome, int novaCarga);
      void ImprimirLista();
      bool EstaVazia();
      int Tamanho();
   }
}
